src/main/java/com/github/manolo8/darkbot/core/objects/facades/QuestProxy.java

```diff
@@ -0,0 +1,300 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.FlashList;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+import com.github.manolo8.darkbot.core.utils.ByteUtils;
+
+import eu.darkbot.api.API;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+public class QuestProxy extends Updatable implements API.Singleton {
+    private @Nullable Quest currentQuest;
+
+    private final List<QuestListItem> questItems = new ArrayList<>();
+
+    private final ObjArray questItemsArr = ObjArray.ofArrObj(true);
+
+    private @Nullable Quest questGiverSelected;
+    private @Nullable QuestListItem questInfoGiverSelected;
+
+    private boolean visibleQuestGiver;
+    private int tabSelected;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        long questClass = API.readMemoryLong(address + 0x98) & ByteUtils.ATOM_MASK;
+
+        this.visibleQuestGiver = API.readMemoryBoolean(address, 0x40);
+        this.tabSelected = API.readMemoryInt(address, 0X50);
+
+        long currentQuestAddr = API.readMemoryLong(questClass + 0x28) & ByteUtils.ATOM_MASK;
+
+        if (currentQuestAddr == 0) {
+            this.currentQuest = null;
+        } else {
+            if (this.currentQuest == null) {
+                this.currentQuest = new Quest();
+            }
+
+            this.currentQuest.update(currentQuestAddr);
+        }
+
+        questItemsArr.update(API.readMemoryPtr(0x58));
+        questItemsArr.sync(questItems, QuestListItem::new);
```

Comment:
i am concerned about the performance implications of this, refreshing a whole long list of quests and all their details every tick can lead to 1000s of mem reads, when most of the times they're not wanted or cared about

If we have it so this only updates in ticks where the getter(s) is/are invoked it would greatly reduce unnecessary overhead

Reply:
Fixed